src/rules/noInvalidThisRule.ts

```diff
@@ -59,39 +59,64 @@ export class Rule extends Lint.Rules.AbstractRule {
 
 function walk(ctx: Lint.WalkContext<boolean>): void {
     const { sourceFile, options: checkFuncInMethod } = ctx;
+
+    const enum parentType {
```

Comment:
Seems semantically odd to recreate this each time `walk` is called. We don't need that, so this should be moved outside of the `walk` function.